### PR TITLE
feat(List composition): expose filter hook

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**
_(This is part of a series of PR to improve api, and help to switch from List legacy component to List Composition)._

Filter on columns works out of the box with a default filter algorithm. But when we want to add some custom filtering, we have to switch to controlled mode, managing the filter state, and the filter of the collection.

**What is the chosen solution to this problem?**
Expose a filter hook, that is used in the List manager.
So once you want to add a custom filter functions
- switch to controlled mode
- use the filter hook, passing a filter function dictionnary

```javascript
const filterFunctions = {
    favorite: (value, textFilter) => { /* filter function on "favorite" data, depending on the textFilter */ }
    quality= (value, textFilter) => { /* filter function on "quality" data, depending on the textFilter */ }
};

const {
    filteredCollection,
    textFilter,
    setTextFilter,
} = useCollectionFilter(collection, initialTextFilter, filterFunctions);
```

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/List/ListComposition/Manager/filter.hook.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { useMemo, useState } from 'react';
+
+function defaultFilterFunction(value, textFilter) {
+	return value
```
Comment (smouillour) [inline packages/components/src/List/ListComposition/Manager/filter.hook.js]: need to test if the value is not null or undefined

File: packages/components/src/List/ListComposition/Manager/filter.hook.test.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		number: 3,
+	},
+	{
+		firstName: 'Carly',
+		lastName: 'Dorsey',
+		number: 4,
+	},
+];
+/*
+	it('should do nothing if no text filter has been provided', () => {
```
Comment (smouillour) [inline packages/components/src/List/ListComposition/Manager/filter.hook.test.js]: if not needed, remove it please

File: output/components.eslint.txt [hunk 3]
```diff
@@ -48,6 +54,6 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/components/src/VirtualizedList/utils/tablerow.js
   41:3  warning  Unexpected console statement  no-console
 
-✖ 24 problems (21 errors, 3 warnings)
-  1 error, 0 warnings potentially fixable with the `--fix` option.
+✖ 26 problems (23 errors, 3 warnings)
+  2 errors, 0 warnings potentially fixable with the `--fix` option.
```
Comment (smouillour) [inline output/components.eslint.txt]: need to fix that

Comment (jsomsanith-tlnd) [reply output/components.eslint.txt]: Fixed one, the other one is due to prettier, so we can't do anything. We should disable indentation linter rules to let prettier as the only source for indentation

File: packages/components/src/List/ListComposition/Manager/filter.hook.test.js (hunk 2 above)
Comment (jsomsanith-tlnd) [reply packages/components/src/List/ListComposition/Manager/filter.hook.test.js]: done

File: packages/components/src/List/ListComposition/Manager/filter.hook.js (hunk 1 above)
Comment (jsomsanith-tlnd) [reply packages/components/src/List/ListComposition/Manager/filter.hook.js]: Set a default value to ''
